infra/build/functions/build_and_run_coverage.py

```diff
@@ -223,14 +251,120 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-arguments
       build_lib.http_upload_step(latest_report_info_body,
                                  latest_report_info_url,
                                  LATEST_REPORT_INFO_CONTENT_TYPE))
+
+  return build_steps
+
+
+def get_fuzz_introspector_steps(  # pylint: disable=too-many-locals, too-many-arguments, unused-argument
+    project_name, project_yaml_contents, dockerfile_lines, image_project,
+    base_images_project, config):
+  """Returns build steps of fuzz introspector for project"""
+  project = build_project.Project(project_name, project_yaml_contents,
+                                  dockerfile_lines, image_project)
+  if project.disabled:
+    logging.info('Project "%s" is disabled.', project.name)
+    return []
+
+  if project.fuzzing_language not in LANGUAGES_WITH_INTROSPECTOR_SUPPORT:
+    logging.info(('Project "%s" is written in "%s", '
+                  'Fuzz Introspector is not supported yet.'), project.name,
+                 project.fuzzing_language)
+    return []
+
+  build_steps = []
+  build = build_project.Build(FUZZING_ENGINE, 'introspector', ARCHITECTURE)
+  env = build_project.get_env(project.fuzzing_language, build)
+
+  report_date = build_project.get_datetime_now().strftime('%Y%m%d')
+  bucket = IntrospectorBucket(project.name, report_date, PLATFORM,
+                              config.testing)
+
+  coverage_report_latest = build_project.get_datetime_yesterday().strftime(
+      '%Y%m%d')
+  # TODO: debug
+  coverage_report_latest = report_date
+  build_steps.append({
+      'args': [
+          'clone', 'https://github.com/google/oss-fuzz.git', '--depth', '1'
+      ],
+      'name': 'gcr.io/cloud-builders/git',
+  })
+
+  bucket_name = 'oss-fuzz-coverage'
+  if config.testing:
+    bucket_name += '-testing'
+
+  coverage_url = (f'{build_lib.GCS_URL_BASENAME}{bucket_name}/{project.name}'
+                  f'/reports/{coverage_report_latest}/linux')
+
+  build_steps.extend(
+      build_lib.download_coverage_data_steps(project.name,
+                                             coverage_report_latest,
+                                             bucket_name, build.out,
+                                             config.testing))
+
+  build_steps.append({
+      'name': 'gcr.io/cloud-builders/docker',
+      'args': ['pull', 'gcr.io/oss-fuzz-base/base-builder:introspector'],
+  })
+  build_steps.append({
+      'name':
+          'gcr.io/cloud-builders/docker',
+      'args': [
+          'tag', 'gcr.io/oss-fuzz-base/base-builder:introspector',
+          'gcr.io/oss-fuzz-base/base-builder:latest'
+      ],
+  })
+
+  build_steps.append({
+      'name': 'gcr.io/cloud-builders/docker',
+      'args': [
+          'build',
+          '-t',
+          f'gcr.io/oss-fuzz/{project.name}',
+          '.',
+      ],
+      'dir': f'oss-fuzz/projects/{project.name}',
```

Comment:
Use posixpath.join instead

Reply:
I though we use posixpath.join for URLs. Did you mean using os.path.join()?

Reply:
Better to use posixpath here, as the Cloud Build environment is posix. The environment this script runs may not be posix.